paasta_tools/cli/cmds/status.py

```diff
@@ -1325,35 +1377,35 @@ def get_replica_state(pod: KubernetesPodV2) -> ReplicaState:
         if not pod.containers or all([c.state == "waiting" for c in pod.containers]):
             state = ReplicaState.STARTING
         else:
-            state = ReplicaState.UNHEALTHY
+            state = ReplicaState.NOT_RUNNING
     elif phase == "Running":
-        if not pod.ready:
-            # TODO: Take sidecar containers into account
-            #   This logic likely needs refining
-            state = ReplicaState.UNREACHABLE
-            main_container = [
-                c
-                for c in pod.containers
-                if c.name not in kubernetes_tools.SIDECAR_CONTAINER_NAMES
-            ]
-            if main_container:
-                if main_container[0].restart_count > 0:
-                    if (
-                        pod.create_timestamp
-                        + main_container[0].healthcheck_grace_period
-                        > datetime.utcnow().timestamp()
-                    ):
-                        state = ReplicaState.WARMING_UP
-                    else:
-                        state = ReplicaState.HEALTHCHECK_FAILING
-                else:
-                    # No restarts yet, still warming up?
-                    state = ReplicaState.WARMING_UP
+        ####
+        #
+        # TODO: Take sidecar containers into account
+        #   This logic likely needs refining
+        main_container = get_main_container(pod)
+        if main_container:
+            warming_up = (
+                pod.create_timestamp + main_container.healthcheck_grace_period
+                > datetime.now().timestamp()
+            )
+            if main_container.state != "running" or not pod.ready:
```

Comment:
> pod.ready means different things depending on whether the pod is in the mesh or not

Ah sorry, this is actually no longer true -- After some feedback a few iterations ago, I changed `pod.ready` to be *strictly* k8s readiness, and now we have `pod.mesh_ready` for *strictly* mesh readiness (based on whether pod ip is in known backends) so we can isolate those conditions a bit better.  But a pod registered to the service mesh *will* still show `pod.ready=False` until the first hacheck readiness probe succeeds. So we'll show NOT_RUNNING until then as well, which I thought was still accurate enough since e.g. wait-for-deployment would also not show a successful bounce until then.  
Do you think showing UNREACHABLE in that case would instead be better?

Reply:
Yeah, I meant that they mean different things, in that the k8s readiness check is different.

Intuitively, I do think that either warming up or unreachable is more accurate. I think this is the most common case (during a bounce) where I'd expect the warming up state.